packages/transport/src/core/Reference.ts

```diff
@@ -6,43 +6,67 @@ import { TransportCoreErrors } from "./TransportCoreErrors";
 
 export interface IReference extends ISerializable {
     id: ICoreId;
+    backboneBaseUrl?: string;
     key: ICryptoSecretKey;
+    forIdentityTruncated?: string;
+    passwordType?: number;
 }
 
 export class Reference extends Serializable implements IReference {
-    @validate()
+    @validate({ regExp: new RegExp("^[A-Za-z0-9]{20}$") })
     @serialize()
     public id: CoreId;
 
+    @validate({ nullable: true })
+    @serialize()
+    public backboneBaseUrl?: string;
+
     @validate()
     @serialize()
     public key: CryptoSecretKey;
 
+    @validate({ nullable: true, regExp: new RegExp("^[0-9a-f]{4}$") })
+    @serialize()
+    public forIdentityTruncated?: string;
+
+    @validate({ nullable: true, min: 1, max: 12, customValidator: (v) => (!Number.isInteger(v) ? "must be an integer" : undefined) })
+    @serialize()
+    public passwordType?: number;
+
     public truncate(): string {
-        const truncatedReference = CoreBuffer.fromUtf8(`${this.id.toString()}|${this.key.algorithm}|${this.key.secretKey.toBase64URL()}`);
+        const idPart = this.backboneBaseUrl ? `${this.id.toString()}@${this.backboneBaseUrl}` : this.id.toString();
+        const truncatedReference = CoreBuffer.fromUtf8(
+            `${idPart}|${this.key.algorithm}|${this.key.secretKey.toBase64URL()}|${this.forIdentityTruncated ? this.forIdentityTruncated : ""}|${this.passwordType ? this.passwordType.toString() : ""}`
+        );
         return truncatedReference.toBase64URL();
     }
 
     public static fromTruncated(value: string): Reference {
         const truncatedBuffer = CoreBuffer.fromBase64URL(value);
         const splitted = truncatedBuffer.toUtf8().split("|");
 
-        if (splitted.length !== 3) {
+        if (![3, 5].includes(splitted.length)) {
             throw TransportCoreErrors.general.invalidTruncatedReference();
         }
 
         try {
```

Comment:
I am not a fan of this try...catch, you should try..catch every parseInt individually to give a better error message